core/opendaq/reader/src/reader_status_impl.cpp

```diff
@@ -54,6 +54,62 @@ ErrCode BlockReaderStatusImpl::getReadSamples(SizeT* readSamples)
     return OPENDAQ_SUCCESS;
 }
 
+TailReaderStatusImpl::TailReaderStatusImpl(const EventPacketPtr& eventPacket, Bool valid, Bool sufficientHistory)
+    : Super(eventPacket, valid)
+    , sufficientHistory(sufficientHistory)
+{
+}
+
+ErrCode TailReaderStatusImpl::getReadStatus(ReadStatus* status)
+{
+    OPENDAQ_PARAM_NOT_NULL(status);
+    if (!sufficientHistory)
+    {
+        *status = ReadStatus::Fail;
+        return OPENDAQ_SUCCESS;
+    }
+    return Super::getReadStatus(status);
+}
+
+ErrCode TailReaderStatusImpl::getSufficientHistory(Bool* status)
+{
+    OPENDAQ_PARAM_NOT_NULL(status);
+    *status = sufficientHistory;
+    return OPENDAQ_SUCCESS;
+}
+
+MultiReaderStatusImpl::MultiReaderStatusImpl(const DictPtr<ISignal, IEventPacket>& eventPackets, Bool valid)
+    : Super(nullptr, valid)
+    , eventPackets(eventPackets)
+{
+}
+
+ErrCode MultiReaderStatusImpl::getReadStatus(ReadStatus* status)
+{
+    OPENDAQ_PARAM_NOT_NULL(status);
+    Bool valid;
+    Super::getValid(&valid);
+
+    if (valid && (!eventPackets.assigned() || eventPackets.getCount() == 0))
+        *status = ReadStatus::Ok;
+    else if (eventPackets.assigned())
+        *status = ReadStatus::Event;
+    else
+        *status = ReadStatus::Fail;
+
+    return OPENDAQ_SUCCESS;
+}
+
+ErrCode MultiReaderStatusImpl::getEventPackets(IDict** events)
+{
+    OPENDAQ_PARAM_NOT_NULL(events);
+    if (eventPackets.assigned())
+        *events = eventPackets.addRefAndReturn();
+    else 
+        *events = Dict<ISignal, IEventPacket>().detach();
```

Comment:
I would create a default empty frozen dictionary to avoid creating it every time